### ISSUE TITLE
Electron-edge Isuue

### ISSUE THREAD

**Issue by bnlakshmibn** [agracio/electron-edge-js#5]
Hi,
I was trying to build an electron app. Using electron-edge for integrating and calling methods from C# class library dll. Referred links https://www.npmjs.com/package/electron-edge-js and https://www.npmjs.com/package/electron-edge.
 I am getting error as "TypeError: Cannot read property 'initializeClrFunc' of undefined".
Using node version 8.9.4.
Stuck with this issue for one week.
Please provide some help if possible.
Thanks

**Comment by agracio** [comment]
Please provide more details on the issue.
* What Electron version you using
* How are you creating your Electron app, are you packaging using any packagers?
* If you have a working Github repo with the issue I will take a look

Local node version is not important for Electron app at it has it own node.dll in package.

**Comment by bnlakshmibn** [comment]
I'm using electron version 1.7.8
I am building electron app via VSCode.

Steps I have done:
 

-  I have downloaded quickstart app from https://github.com/electron/electron-quick-start.

- Installed electron-edge (npm install electron-edge)

- Electron version is 1.7.8 and node version is 8.9.4, electron edge version 6.5.5

- Created a .net standard 2.0 class library and added the dll to the app folder

- Added a sample.js file in to my app folder

 var edge = require('electron-edge');
var clrMethod = edge.func({
    assemblyFile: 'EdgeSample.dll',
    typeName: 'EdgeSample.Sample',
    methodName: 'Conversion' // This must be Func<object,Task<object>> 
});
clrMethod('Some input', function (error, result) {
    if (error) throw error;
    console.log(result);
});
- executed the command node sample.js

- First got the error " The edge module has not been pre-compiled for node.js version v8.9.4. You must build a custom version of edge.node"

- So installed node version 7.4.0

- Again ran the same command node sample.js

- Then got the error " TypeError: Cannot read property 'initializeClrFunc' of undefined"

Sorry I dont have a working repo for this.

**Comment by agracio** [comment]
`electron-edge-js` can only be used inside Electron app, you cannot use it with a regular Node.js code that you launch separately. Your code should be executed as part of Electron app itself then you will see output of your `console.log(result)` statement in Electron _Developer Tools_ window. See [this issue](https://github.com/agracio/electron-edge-js/issues/6) for more information.

**Comment by bnlakshmibn** [comment]
I have used electron-edge-js and it worked fine. But yet another issue is there in using .net standard class library from electron edge.

I have created a sample electron app and installed electron-edge-js package. Tried to add a .net framework class library dll and called a method from it. It worked fine.

But the issue is, I tried to create a .net standard 2.0 class library and referenced that dll in my app and tried calling a method from that (a method which simply returns a string). But I got the error " 'Could not load file or assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.'"
